test/Response/SapiStreamEmitterTest.php

```diff
@@ -57,76 +61,322 @@ public function testDoesNotInjectContentLengthHeaderIfStreamSizeIsUnknown()
+            if (! is_array($item)) {
+                $item = [$item];
+            }
+
+            $parameters['items'] = array_merge($parameters['items'], $item);
+
+            if (count($parameters['initialData']) > 0) {
+                array_walk(array_shift($parameters['initialData']), $closureExpand, $parameters);
+            } else {
+                $parameters['resultData'][] = $parameters['items'];
+            }
+        };
+
+        array_walk(array_shift($initialData), $closureExpand, [
+            'initialData' => $initialData,
+            'items' => [],
+            'resultData' => & $resultData]);
+
+        return $resultData;
+    }
+
+    private function getStreamProphecy(& $contents, & $size, & $position, & $peakBufferLength)
+    {
+        $stream = $this->prophesize('Psr\Http\Message\StreamInterface');
+
+        $stream->__toString()->will(function () use (& $contents, & $position, $size) {
+            if (is_callable($contents)) {
+                $data = $contents(0);
+            } else {
+                $data = $contents;
+            }
+
+            $position = $size;
+
+            return $data;
+        });
+
+        $stream->getSize()->willReturn($size);
+
+        $stream->tell()->will(function () use (& $position) {
+            return $position;
+        });
+
+        $stream->eof()->will(function () use ($size, & $position) {
+            return ($position >= $size);
+        });
+
+        $stream->seek(Argument::type('integer'), Argument::any())->will(function ($args) use ($size, & $position) {
+            if ($args[0] < $size) {
+                $position = $args[0];
+                return true;
+            }
+
+            return false;
+        });
+
+        $stream->rewind()->will(function () use (& $position) {
+            $position = 0;
+            return true;
+        });
+
+        $stream->read(Argument::type('integer'))
+            ->will(function ($args) use (& $contents, & $position, & $peakBufferLength) {
+                if (is_callable($contents)) {
+                    $data = $contents($position, $args[0]);
+                } else {
+                    $data = substr($contents, $position, $args[0]);
+                }
+
+                if ($args[0] > $peakBufferLength) {
+                    $peakBufferLength = $args[0];
+                }
+
+                $position += strlen($data);
+
+                return $data;
+            });
+
+        $stream->getContents()->will(function () use (& $contents, & $position) {
+            if (is_callable($contents)) {
+                $remainingContents = $contents($position);
+            } else {
+                $remainingContents = substr($contents, $position);
+            }
+
+            $position += strlen($remainingContents);
+
+            return $remainingContents;
+        });
+
+        return $stream;
+    }
+
+    public function emitStreamResponseProvider()
```

Comment:
@Ocramius  
I'm not sure if after the changes this still necessary.
Also, I don't know exactly how to document elements of an array according to the ZF CS (as far as I know. There is no official way to do this in phpdoc).